building_dialouge_webapp/heat/flows.py

```diff
@@ -0,0 +1,964 @@
+    ):
+        super().__init__(flow, name, form_class, template_name, label)
+        self.info_text = info_text
+
+    @property
+    def response(self) -> dict[str, StateResponse]:
+        form_response = {
+            "info": HTMLStateResponse(
+                f'<div id="info" hx-swap-oob="innerHTML">{self.info_text}</div>',
+            ),
+        }
+        form_response.update(**super().response)
+        return form_response
+
+    @property
+    def reset_response(self) -> dict[str, StateResponse]:
+        form_response = {
+            "info": HTMLStateResponse('<div id="info" hx-swap-oob="innerHTML"></div>'),
+        }
+        form_response.update(**super().reset_response)
+        return form_response
+
+
+class Transition:
+    def __init__(self):
+        # These are set once Transition is added to state
+        self.flow = None
+
+    @abstractmethod
+    def follow(self, state: "State") -> "State":
+        """Must be implemented by subclasses."""
+
+    def _state(self, state_name) -> "State":
+        return getattr(self.flow, state_name)
+
+
+class Next(Transition):
+    def __init__(self, next_state: str):
+        super().__init__()
+        self.next_state = next_state
+
+    def follow(self, state: "State") -> "State":
+        return self._state(self.next_state)
+
+
+class Switch(Transition):
+    """
+    Attributes:
+        lookup (str): the field of the corresponding form on wich the switch depends
+        cases : .case("returned value of the field", "name of next state")
+    """
+
+    def __init__(self, lookup: str | Callable | None = None):
+        super().__init__()
+        self.lookup = lookup
+        self.cases = {}
+
+    def case(self, value: Any, state_name: str) -> "Switch":
+        self.cases[value] = state_name
+        return self
+
+    def default(self, state_name: str) -> "Switch":
+        self.cases["_default"] = state_name
+        return self
+
+    def follow(self, state: "State") -> "State":
+        result = (
+            self.lookup(state)
+            if self.lookup is not None and not isinstance(self.lookup, str)
+            else self.default_switch_fct(state)
+        )
+        if result in self.cases:
+            return self._state(self.cases[result])
+        if "_default" in self.cases:
+            return self._state(self.cases["_default"])
+        error_msg = f"No option for result '{result}' found, no default given."
+        raise FlowError(error_msg)
+
+    def default_switch_fct(self, state: "State") -> Any:
+        key = self.lookup if isinstance(self.lookup, str) else state.name
+        session_data = self.flow.request.session.get("django_htmx_flow", {})
+        if key in session_data:
+            return session_data[key]
+        if key in self.flow.request.POST:
+            return self.flow.request.POST[key]
+        error_msg = f"Could not find lookup {key=} in request or session."
+        raise FlowError(error_msg)
+
+
+class Flow(TemplateView):
```

Comment:
solved this by using the object in the get_flows() structure to compare where in the list the current object calling the index is. and depending on that the ones before get visited and the ones after get empty, and the current object gets active.
It helps because this way views can also use this.